### PR TITLE
Updated apps types to reflect server changes

### PR DESCRIPTION
**Pull request by levb**
Matches https://github.com/mattermost/mattermost-plugin-apps/pull/32

### CODE REVIEW

**Review by jfrerich — APPROVED** [review]

**Review by mickmister — CHANGES_REQUESTED** [review]

File: src/selectors/entities/apps.ts [hunk 1]
```diff
@@ -5,9 +5,10 @@ import {AppBinding} from 'types/apps';
 
 export function getAppsBindings(state: GlobalState, location?: string): AppBinding[] {
     if (location) {
-        return state.entities.apps.bindings.filter((p) => {
-            return p.location_id === location;
-        });
+        const headerBindings = state.entities.apps.bindings.find((b) => b.location_id === location);
+        if (headerBindings && headerBindings.bindings) {
+            return headerBindings.bindings!;
```
[the remark this answers is not in the document]
Comment (levb) [reply src/selectors/entities/apps.ts:10]: Please make it right... I was getting a TS lint error.

[the remark this answers is not in the document]
Comment (mickmister) [reply src/selectors/entities/apps.ts:10]: It shows up elsewhere in the codebase. Puzzled by this, as the javascript output is identical, except there is no exclamation point. (also I apologize for seemingly yelling in my above comment)

[the remark this answers is not in the document]
Comment (mickmister) [reply src/selectors/entities/apps.ts:10]: Circling back to this, what is the main benefit of having the bindings structured this way, as opposed to a flat list? The level of indirection here versus of the data flow from `server -> mattermost-redux -> mattermost-webapp` seems off. Why should the webapp expect to receive data that is not the same form that came from the server?

**Review by mickmister — APPROVED** [review]

File: src/constants/apps.ts [hunk 2]
```diff
@@ -4,4 +4,5 @@ export default {
     APPS_BINDINGS_POST_MENU_ITEM: '/post_menu',
     APPS_BINDINGS_CHANNEL_HEADER_ICON: '/channel_header',
     APPS_BINDINGS_COMMAND: '/command',
+    APPS_BINDINGS_IN_POST: '/in_post',
```
Comment (larkox) [inline src/constants/apps.ts:7]: Random idea: Are we considering more stuff to add "in_post" apart of embedded forms? I cannot think of any useful idea, but I still see potential.

File: src/types/apps.ts [hunk 3]
```diff
@@ -35,30 +40,38 @@ export type AppBinding = {
...
-export const AppCallResponseTypes: {[name: string]: AppCallResponseType} = {
-    OK: 'ok',
+export const AppCallResponseTypes: { [name: string]: AppCallResponseType } = {
+    OK: '',
     ERROR: 'error',
-    MODAL: 'modal',
-    NAVIGATE: 'navigate',
+    FORM: 'form',
     CALL: 'call',
-    COMMAND: 'command',
```
Comment (larkox) [inline src/types/apps.ts:61]: Are we dropping the possibility to run "any command"?

**Review by larkox — APPROVED** [review]
LGTM, just one small question.

**Pull request merged**